village/src/App.js

```diff
@@ -1,27 +1,50 @@
-import React, { Component } from 'react';
-
-import './App.css';
-import SmurfForm from './components/SmurfForm';
-import Smurfs from './components/Smurfs';
+import React, { Component } from "react";
+import axios from "axios";
+import { Route, NavLink } from "react-router-dom";
+import "./App.css";
+import SmurfForm from "./components/SmurfForm";
+import Smurfs from "./components/Smurfs";
 
 class App extends Component {
   constructor(props) {
     super(props);
     this.state = {
-      smurfs: [],
+      smurfs: []
     };
   }
   // add any needed code to ensure that the smurfs collection exists on state and it has data coming from the server
   // Notice what your map function is looping over and returning inside of Smurfs.
   // You'll need to make sure you have the right properties on state and pass them down to props.
+  componentDidMount() {
+    this.fetchSmurfs();
+  }
+  fetchSmurfs = () => {
+    axios.get('http://localhost:3333/smurfs')
+      .then(res => this.setState({ smurfs: res.data }))
+      .catch(err => console.log(err));
+  }
+
   render() {
     return (
       <div className="App">
-        <SmurfForm />
-        <Smurfs smurfs={this.state.smurfs} />
+        <nav>
+          <NavLink to="/">Hide Add Smurf</NavLink>
+          </nav>
+        <nav>
+          <NavLink to="/smurf-form">Show Add Smurf</NavLink>
+        </nav>
+
+        <Route
+          path="/smurf-form"
+          render={props => <SmurfForm {...props} getSmurfs={this.getSmurfs} />}
```

Comment:
You don't have a `getSmurfs` method on this component. So, you're passing an `undefined` props here.